src/main/scala/org/globalforestwatch/layers/Layer.scala

```diff
@@ -145,8 +146,12 @@ trait RequiredLayer extends Layer {
   lazy val source: GeoTiffRasterSource = {
     // Removes the expected 404 errors from console log
     val s3uri = new AmazonS3URI(uri)
-    if (!s3Client.doesObjectExist(s3uri.getBucket, s3uri.getKey)) {
-      throw new FileNotFoundException(uri)
+    try {
+      if (!s3Client.doesObjectExist(s3uri.getBucket, s3uri.getKey)) {
+        throw new FileNotFoundException(uri)
+      }
+    } catch {
+      case e: AmazonS3Exception => throw new FileNotFoundException(uri)
```

Comment:
How come the check above is not sufficient? What other S3 Exceptions were thrown?

Reply:
Because doesObjectExist uses a head object call to check. That call requires ListBucket permissions, but we only publicly give GetObject permissions on the bucket. See under Permissions here: https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html

"You need the s3:GetObject permission for this operation. For more information, see Specifying Permissions in a Policy. If the object you request does not exist, the error Amazon S3 returns depends on whether you also have the s3:ListBucket permission.

If you have the s3:ListBucket permission on the bucket, Amazon S3 returns an HTTP status code 404 ("no such key") error.

If you don’t have the s3:ListBucket permission, Amazon S3 returns an HTTP status code 403 ("access denied") error."

I'm assuming we're not giving out public ListBucket access for security reasons, right?

Reply:
ok, got it. 

We should probably update our bucket policy and add ListBucket permissions. 
Ideally, I'd like to make the all public, but with requester paid option.

If I remember correctly, we didn't do that yet, b/c of geotrellis 2 restrictions. This restrictions aren't there anymore in geotrellis 3, right? So maybe after the upgrade we can work on that.

Reply:
yeah, I think it's fixed once we moved to use GDAL with geotrellis, since GDAL has an environment option to request payer.